src/jsSSE.php

```diff
@@ -107,6 +128,15 @@ private function output(string $content): void
      */
     public function sendBlock(string $id, string $data, string $name = null): void
     {
+        if (connection_aborted()) {
+            $this->hook('aborted');
```

Comment:
Add `public const HOOK_ABORTED = 'aborted';` and use it here like `$this->hook(static::HOOK_ABORTED);`. Then remove `onAborted()` method.

Reply:
agree to add constants for every classes that have hooks i think adding a detailed description, i suggest to add callback parameters and what happens if a value is returned.
HOOK_{hook_name}

i don't remember if sphinx can do it but if we write correct docblocks, we can inject in documentation directly to avoid write it 2 times.

Reply:
I really do not see where is the problem with the onAborted() function. It is just a helper function for the developer. We will probably use this function a lot more for Saasty. An external developer may use this for installing addon to Saassty for example, so I think it is important. Same as onSubmit() is with form.

Your suggestion with static:: HOOK_ABORTED is good but and we should probably do it in a separate pr for all hook as well.